core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1800,21 +1804,26 @@ class SparkContext(config: SparkConf) extends Logging {
         addJarFile(new File(path))
       } else {
         val uri = new URI(path)
-        // SPARK-17650: Make sure this is a valid URL before adding it to the list of dependencies
         Utils.validateURL(uri)
         uri.getScheme match {
-          // A JAR file which exists only on the driver node
+            //  A JAR file which exists only on the driver node
           case null =>
-            // SPARK-22585 path without schema is not url encoded
+            //  SPARK-22585 path without schema is not url encoded
             addJarFile(new File(uri.getRawPath))
-          // A JAR file which exists only on the driver node
-          case "file" => addJarFile(new File(uri.getPath))
-          // A JAR file which exists locally on every worker node
+          case "file" => addJarFile(new File(uri.getRawPath))
```

Comment:
Why here change to `getRawPath`?

Reply:
@jerryshao Make a mistake,I will change back

Reply:
Done